kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/net/EndpointRegistry.java

```diff
@@ -618,6 +618,28 @@ public CompletionStage<Void> shutdown() {
         return allOfStage(registeredVirtualClusters.keySet().stream().map(this::deregisterVirtualCluster));
     }
 
+    /**
+     * Selects a bootstrap server from the list using a round-robin approach based on the current time.
+     * This distributes load across multiple bootstrap servers and provides better resilience when one server is down.
+     * 
+     * @param bootstrapServers the list of bootstrap servers
+     * @return the selected bootstrap server
+     */
+    private HostPort selectBootstrapServer(List<HostPort> bootstrapServers) {
+        if (bootstrapServers == null || bootstrapServers.isEmpty()) {
+            throw new IllegalArgumentException("Bootstrap servers list cannot be null or empty");
+        }
+        
+        if (bootstrapServers.size() == 1) {
+            return bootstrapServers.get(0);
+        }
+        
+        // Use current time as a source of variation to distribute across servers
+        // This provides better distribution than always using the first server
+        int index = (int) (System.currentTimeMillis() % bootstrapServers.size());
```

Comment:
Using time will make the system non-deterministic, which makes the unit testing tricky.  Would a round-robin approach work for your use-case?

Reply:
I guess for now, simple round-robin should suffice for our use case. Let me update it using simple round robin.

Reply:
the EndpointRegistry is shared by the whole proxy, so if the round robining is scoped to the EndpointRegistry, then you'll get seemingly odd behaviour in the case where there are multiple virtualclusters.   I think it'd be better to the roundrobining was done per targetcluster which would avoid this behaviour.

So I think it would be better if EndpointRegistry had a:

```java
    private final Map<TargetCluster, AtomicInteger> counters = new ConcurrentHashMap<>();
```

then use something like;

```java
counter = counters.compute(targetCluster, (k, v) -> (v == null) ? new AtomincLong(0) : v.getAndIncrement())}
```